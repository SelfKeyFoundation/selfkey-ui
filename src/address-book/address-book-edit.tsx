import * as React from 'react';
import { Grid, WithStyles, withStyles, createStyles } from '@material-ui/core';
import StyledButton from '../common/styled-button';

const styles = createStyles({
    errorText: {
        height: '19px',
        width: '242px',
        color: '#FE4B61',
        fontFamily: 'Lato',
        fontSize: '13px',
        lineHeight: '19px'
    },
    errorColor: {
        color: '#FE4B61 !important',
        border: '2px solid #FE4B61 !important',
        backgroundColor: 'rgba(255,46,99,0.09) !important'
    },
    input: {
        boxSizing: 'border-box',
        height: '46px',
        width: '722px',
        border: '1px solid #384656',
        borderRadius: '4px',
        backgroundColor: '#1E262E',
        color: '#a9c5d6',
        fontSize: '14px',
        boxShadow: 'inset -1px 0 0 0 rgba(0,0,0,0.24), 1px 0 0 0 rgba(118,128,147,0.2), 2px 0 2px 0 rgba(0,0,0,0.2)',
        paddingLeft: '10px',
        '&::-webkit-input-placeholder': {
            fontSize: '14px',
            color: '#93B0C1',
        }
    },
    inputError: {
        borderBottom: '2px solid #FE4B61 !important'
    },
    label: {
        color: '#93A4AF',
        fontSize: '12px',
        fontWeight: 'bold',
        lineHeight: '15px'
    }
});

export type AddressBookEditState = {
    label: string
};

export type AddressBookEditProps = {
    label: string,
    onSave: (label: string) => void,
    labelError?: string,
    onCancel: () => void,
    onLabelChange: (label: string) => void,
}

export type AddressBookEditStyledProps = WithStyles<typeof styles> & AddressBookEditProps;

export class AddressBookEditComponent extends React.Component<AddressBookEditStyledProps, AddressBookEditState> {

    constructor(props: AddressBookEditStyledProps) {
        super(props);
        this.state = {label: props.label};
    }

    componentDidUpdate() {
        // this.state = {label: this.props.label};
    }

    handleLabelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        event.preventDefault();
        const label = event.target.value;
        this.setState({...this.state, label});
        this.props.onLabelChange(label);
    };

    handleSubmit = (event: any) => {
        event.preventDefault();
        return this.props.onSave(this.state.label);
    }

    render() {
        const { classes, labelError, onCancel} = this.props;
        const hasLabelError = (labelError !== '' && labelError !== undefined)
        const labelInputClass = `${classes.input} ${hasLabelError ? classes.errorColor : ''}`;
        return (
            <form id="formwrap" noValidate autoComplete="off" onSubmit={this.handleSubmit}>
                <Grid container direction='column' spacing={10}>
                    <Grid item>
                        <Grid container direction='column' spacing={1}>
                            <Grid item>
                                <label className={classes.label}>LABEL</label>
                            </Grid>
                            <Grid item>
                                <input type='text' id='labelInput' onChange={this.handleLabelChange} value={this.state.label} className={labelInputClass} placeholder="Address label" />
                                {hasLabelError && <span id='labelError' className={classes.errorText}>{labelError}</span>}
                            </Grid>
                        </Grid>
                    </Grid>
                    <Grid item>
                        <Grid container direction='row' spacing={8}>
                            <Grid item>
                                <StyledButton id='saveButton' variant="contained" size="medium" type="submit" disabled={(!this.state.label || hasLabelError)}>
                                    Save
                                </StyledButton>
                            </Grid>
                            <Grid item>
                                <StyledButton id='cancelButton' variant="outlined" size="medium" onClick={onCancel}>
                                    Cancel
                                </StyledButton>
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
            </form>
        );
    }
}

export const AddressBookEdit = withStyles(styles)(AddressBookEditComponent);

export default AddressBookEdit;
