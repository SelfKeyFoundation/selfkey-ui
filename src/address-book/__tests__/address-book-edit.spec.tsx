import * as React from 'react';
import { mount } from 'enzyme';
import { AddressBookEdit } from '../address-book-edit';

describe('Address Book Add', () => {
  it('should show Address Book Add', () => {
    const component = mount(
      <AddressBookEdit
        label='Test'
        onSave={(label) => alert('onEdit' + label)}
        onCancel={() => alert('onCancel')}
        onLabelChange={label => alert('onLabelChange ' + label)}
        labelError='hey'
      />
    );
    expect(component).toBeDefined();
  });

  it('should trigger onChange', () => {
    const onLabelChangeMock = jest.fn();
    const component = mount(
      <AddressBookEdit
        label='Test'
        onSave={() => console.log('onEdit')}
        onCancel={() => console.log('onCancel')}
        onLabelChange={onLabelChangeMock}
        labelError='hey'
      />);
      component.find('#labelInput').simulate('change', { target: { value: 'x' }});
      // expect(component.find('#labelInput').prop('value')).toEqual('x');
      expect(onLabelChangeMock).toBeCalled();

      // component.setState({label: component.find('#labelInput').prop('value')});
      expect(component.find('#formwrap').simulate('submit')).toBeDefined();
  });

  it('should trigger Submit', () => {
    const component = mount(
      <AddressBookEdit
        label='Test'
        onSave={() => console.log('onEdit')}
        onCancel={() => console.log('onCancel')}
        onLabelChange={() => console.log('onLabelChange')}
        labelError='hey'
      />);
      // component.setState({label: component.find('#labelInput').prop('value')});
      expect(component.find('#formwrap').simulate('submit')).toBeDefined();
  });
});





