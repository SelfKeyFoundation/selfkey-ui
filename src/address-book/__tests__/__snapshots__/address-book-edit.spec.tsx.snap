// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Book Add should show Address Book Add 1`] = `
<form
  autoComplete="off"
  className={undefined}
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-32-26 MuiGrid-direction-xs-column-4"
  >
    <div
      className="MuiGrid-item-2"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-direction-xs-column-4"
      >
        <div
          className="MuiGrid-item-2"
        >
          <label
            className="AddressBookEditComponent-label-0-1-5"
          >
            LABEL
          </label>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <input
            className="AddressBookEditComponent-input-0-1-3 AddressBookEditComponent-errorColor-0-1-2"
            id="labelInput"
            onChange={[Function]}
            placeholder="Address label"
            type="text"
            value="Test"
          />
          <span
            className="AddressBookEditComponent-errorText-0-1-1"
            id="labelError"
          >
            hey
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
      >
        <div
          className="MuiGrid-item-2"
        >
          <button
            className="MuiButtonBase-root-124 MuiButtonBase-disabled-125 MuiButton-root-98 Component-root-0-1-6 MuiButton-contained-109 MuiButton-containedPrimary-110 Component-containedPrimary-0-1-7 MuiButton-raised-112 MuiButton-raisedPrimary-113 MuiButton-disabled-118 Component-disabled-0-1-9"
            disabled={true}
            id="saveButton"
            onBlur={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="-1"
            type="submit"
          >
            <span
              className="MuiButton-label-99"
            >
              Save
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-item-2"
        >
          <button
            className="MuiButtonBase-root-124 MuiButton-root-98 Component-root-0-1-6 MuiButton-text-100 MuiButton-textPrimary-101 MuiButton-outlined-106 Component-outlined-0-1-8 MuiButton-outlinedPrimary-107"
            disabled={false}
            id="cancelButton"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-99"
            >
              Cancel
            </span>
            <span
              className="MuiTouchRipple-root-127"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
